small-talk/app/homepage/page.js

```diff
@@ -1,156 +1,71 @@
+                </button>
+                <button onClick={handleRedirect('/profile')}>
+                    <img src="/img/profile-temp.png" alt="Profile" style={{ height: '115px', marginTop: '10px', marginBottom: '10px', marginLeft: '20px' }}/>
+                </button>
+            </div>
+            <div className="sidebar">
+                {/* creating a health button, and direct to the health page with ButtonStyling for the button */}
+                <HoverButton onClick={handleRedirect('/health')} iconImage="/img/health-tab.png" altText="Health" />
+
+                {/* creating a events button, and direct to the events page with ButtonStyling for the button */}
+                <HoverButton onClick={handleRedirect('/events')} iconImage="/img/event-tab.png" altText="Events" />
 
                 {/* creating a chat button, and direct to the chat page with ButtonStyling for the button */}
-                <HoverButton style={ButtonStyle} onClick={handleChat}>Chat</HoverButton>
+                <HoverButton onClick={handleRedirect('/chat')} iconImage="/img/chat-tab.png" altText="Chat" />
 
-                {/* creating a food button, and direct to the food page with ButtonStyling for the button */}
-                <HoverButton style={ButtonStyle} onClick={handleFood}>Order Food</HoverButton>
+                {/* creating a chat button, and direct to the chat page with ButtonStyling for the button */}
+                <HoverButton onClick={handleRedirect('/friends')} iconImage="/img/friend-tab.png" altText="Friends" />
 
                 {/* creating a movies button, and direct to the movies page with ButtonStyling for the button */}
-                <HoverButton style={ButtonStyle} onClick={handleMovies}>Movies</HoverButton>
+                <HoverButton onClick={handleRedirect('/movie')} iconImage="/img/media-tab.png" altText="Movies" />
 
-                {/* creating a settings button, and direct to the settings page with ButtonStyling for the button */}
-                <HoverButton style={ButtonStyle} onClick={handleSettings}>Settings</HoverButton>
-
-                {/* creating a profile button, and direct to the profile page with ButtonStyling for the button */}
-                <HoverButton style={ButtonStyle} onClick={handleProfile}>Profile</HoverButton>
+                {/* creating a food button, and direct to the food page with ButtonStyling for the button */}
+                <HoverButton onClick={handleRedirect('/food')} iconImage="/img/food-tab.png" altText="Order Food" />
 
-                {/*creating a signout button, and direct to the login page with ButtonStyling for the button */}
-                <HoverButton style={ButtonStyle} onClick={handleSignOut}>Sign Out</HoverButton>
+                {/* creating a game button, and direct to the game page with ButtonStyling for the button */}
+                <HoverButton onClick={handleRedirect('/game')} iconImage="/img/games-tab.png" altText="Games" />
```

Comment:
Would love if we could implement a <menuButton> custom component that can take a list of redirects, icon image paths, and altText strings and map the list to a series of menu buttons. Then we can be sure to map that within a sidebar custom component as well to separate the functionality easier.

Reply:
This is being handled in issue #52